src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+    faceToElements[curFace][1] = neighborCellInMaybeNotSameBlock;
+  }
+  return true;
+}
+
+// Using a set means bad performances and annoying code 
+// TODO Change set to vector
+void HexMeshConnectivityBuilder::getFacesAroundEdge( 
+  int edgeIndex, 
+  std::vector< localIndex > const & allFacesToUniqueFace,
+  std::set <localIndex> & faces ) const
+{
+  faces.clear();
+
+  int first = m_uniqueEdges[edgeIndex];
+  int last = m_allEdges.size();
+
+  if( edgeIndex+1 < nbEdges )
+  {
+    last = m_uniqueEdges[edgeIndex+1];
+  }
+
+  // For each duplicate of this edge
+  for( int i = first; i < last; ++i)
+  {
+    LocalEdgeIndex id = m_allEdges[i].second;
+    // Get the cell
+    LocalCellIndex c = id / Hex::nbEdges;
+    // Get the hex facet incident to the edge
+    HexEdgeIndex e = id % Hex::nbEdges;
+    HexFacetIndex f0 = Hex::edgeAdjacentFacet[e][0];
+    HexFacetIndex f1 = Hex::edgeAdjacentFacet[e][1];
+    // Get the face index in allFaces 
+    LocalFaceIndex face0 = Hex::nbFacets * c + f0;
+    LocalFaceIndex face1 = Hex::nbFacets * c + f1;
+    // Get the uniqueFaceId 
+    localIndex uniqueFace0 = allFacesToUniqueFace[face0];
+    localIndex uniqueFace1 = allFacesToUniqueFace[face1];
+
+    faces.insert(uniqueFace0);
+    faces.insert(uniqueFace1);
+  }
+}
+
+
+bool HexMeshConnectivityBuilder::computeFacesToEdges( ArrayOfArrays<localIndex> & faceToEdges) const 
+{
+  // 1 - Allocate
+  faceToEdges.resize(0);
+  // TODO Check if this resize or reserve the space for the inner arrays
+  faceToEdges.resize( nbFaces, Hex::nbEdgesPerFacet); 
+  // In doubt resize inner arrays
+  for (int i = 0; i < nbFaces; ++i)
+  {
+    faceToEdges.resizeArray(i, Hex::nbEdgesPerFacet );
+    // Initialize the values - Is there a simpler way to do this?
+    faceToEdges[i][0] = NO_ID;
+    faceToEdges[i][1] = NO_ID;
+    faceToEdges[i][2] = NO_ID;
+    faceToEdges[i][3] = NO_ID;
+  }
+
+  std::vector< localIndex > allFacesToUniqueFace;
+  computeAllFacesToUniqueFace( allFacesToUniqueFace );
+
+  for (unsigned int edgeIndex = 0; edgeIndex < m_uniqueEdges.size(); ++edgeIndex)
+  {
+    std::set<localIndex> faces;
+    getFacesAroundEdge( edgeIndex, allFacesToUniqueFace, faces );
+
+    for( auto itr( faces.begin()); itr !=faces.end(); ++itr)
+    { 
+      if     ( faceToEdges[*itr][0] == NO_ID )  faceToEdges[*itr][0] = edgeIndex ;
+      else if( faceToEdges[*itr][1] == NO_ID )  faceToEdges[*itr][1] = edgeIndex ;
+      else if( faceToEdges[*itr][2] == NO_ID )  faceToEdges[*itr][2] = edgeIndex ;
+      else if( faceToEdges[*itr][3] == NO_ID )  faceToEdges[*itr][3] = edgeIndex ;
+      else assert( false ); 
+    }
+  }
+  return true;
+}
+
+bool HexMeshConnectivityBuilder::computeEdgesToFaces(ArrayOfSets<localIndex> & edgeToFaces ) const
+{
+  // 1 - Allocate - Without counting  -- Should we overallocate a bit ?
+  // TODO Count for tet meshes one should count
+  int numberOfFacesAroundEdge = 5; // In a hexahedral mesh this is enough
```

Comment:
When you say it's enough, do you mean that it's mathematical of that it should be OK for most cases?

Reply:
There is not theoretical limit, it should be OK for most cases. A fortiori for regular meshes 4 is the limit.